.github/workflows/ci.yml

```diff
@@ -43,40 +43,53 @@ jobs:
       - name: basedpyright
         run: uv run poe pyright
 
-  typetest:
-    timeout-minutes: 5
+  generate-matrix:
+    name: Generate Test Matrix
     runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
-        uses: astral-sh/setup-uv@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
 
-      - name: install
-        run: uv sync --frozen --python 3.13
-
-      - name: typetest
-        run: uv run poe typetest
+      - name: Install Python Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install scipy
 
-      - name: stubtest
-        run: uv run poe stubtest
+      - name: Generate Matrix
+        id: set-matrix
+        shell: bash
+        run: |
+          matrix=$(python scripts/generate_matrix.py | jq -c)
```

Comment:
out of curiosity; what does the `jq -c` do here?

Reply:
Github actions has some shell magic that breaks down from the matrix. The `jq -c` command condenses the json so it doesn't break things. Also in the case of multiline json which can be convenient for debugging :)